pkg/trustengine/trust_engine.go

```diff
@@ -0,0 +1,62 @@
+package trustengine
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+type trustEngineUtils interface {
+	*piperhttp.Client
+}
+
+type TrustEngineSecret struct {
+	Token string `json:"sonar,omitempty"`
+}
```

Comment:
Export a tool-agnostic struct

```suggestion
type TrustEngineSecret struct {
	Name string `json:"name"`
	Token string `json:"token"`
}
```

Reply:
Done